src/main/java/org/opensearch/ad/feature/AbstractRetriever.java

```diff
@@ -30,8 +32,49 @@
 import org.opensearch.search.builder.SearchSourceBuilder;
 
 public abstract class AbstractRetriever {
-    protected double parseAggregation(Aggregation aggregation) {
+    protected double parseAggregation(Aggregation aggregationToParse) {
         Double result = null;
+        /* example InternalSingleBucketAggregation: filter aggregation like
+           "t_shirts": {
+                "filter": {
+                    "bool": {
+                        "should": [
+                            {
+                                "term": {
+                                    "issueType": "foo"
+                                }
+                            }
+                            ...
+                            ],
+                            "minimum_should_match": "1",
+                            "boost": 1
+                    }
+                },
+                "aggs": {
+                    "impactUniqueAccounts": {
+                        "aggregation": {
+                            "field": "account"
+                        }
+                    }
+                }
+            }
+        
+            would produce an InternalFilter (a subtype of InternalSingleBucketAggregation) with a sub-aggregation
+            InternalCardinality that is also a SingleValue
+        */
+
+        if (aggregationToParse instanceof InternalSingleBucketAggregation) {
+            InternalAggregations bucket = ((InternalSingleBucketAggregation) aggregationToParse).getAggregations();
+            if (bucket != null) {
+                List<Aggregation> aggrs = bucket.asList();
+                if (aggrs.size() == 1) {
+                    // we only accept a single value as feature
+                    aggregationToParse = aggrs.get(0);
```

Comment:
So if any of the 3 "if" conditions fail, it would fall back to the previous logic and still may fail with EndRunException: Failed to parse aggregation, right?

Reply:
yes